bynary/apps/demo/src/app/core/feature/recipe/pages/detail/recipe-detail.component.spec.ts

```diff
@@ -138,43 +140,80 @@ describe('RecipeDetailComponent', () => {
                 expect(recipeDescription.nativeElement.textContent).toEqual(' Bla bla ');
             });
 
-            it('should have a ingredients', () => {
+            it('should have ingredients', () => {
                 const recipeIngredients = debugElement.queryAll(By.css('.c-recipe-detail-component__recipe-ingredients'));
 
                 expect(recipeIngredients).toHaveLength(2);
-                expect(recipeIngredients[0].nativeElement.textContent).toEqual(' Schnitzel - 2 ');
-                expect(recipeIngredients[1].nativeElement.textContent).toEqual(' Pommes - 1 ');
             });
 
-        });
+            describe('ingredients', () => {
+                let recipeIngredients: DebugElement[];
 
-        // TODO: Überprüfen, ob Buttons richtige Texte anzeigen
+                beforeEach(() => {
+                    recipeIngredients = debugElement.queryAll(By.css('.c-recipe-detail-component__recipe-ingredients'));
+                });
 
-        it('should call onAddToShoppingList method when button "To Shopping List" is clicked', () => {
-            const addToShoppingList = jest.spyOn(component, 'onAddToShoppingList');
-            const toShoppingListButton = debugElement.query(By.css('#toShoppingList'));
+                it('should contain ingredient Schnitzel (2)', function () {
+                    expect(recipeIngredients[0].nativeElement.textContent).toEqual(' Schnitzel - 2 ');
+                });
 
-            toShoppingListButton.triggerEventHandler('click', addToShoppingList);
+                it('should contain ingredient Pommes (1)', function () {
+                    expect(recipeIngredients[1].nativeElement.textContent).toEqual(' Pommes - 1 ');
+                });
+
+            });
 
-            expect(addToShoppingList).toHaveBeenCalledTimes(1);
         });
 
-        it('should call onEditRecipe method when button "Edit Recipe" is clicked', () => {
-            const onEditRecipe = jest.spyOn(component, 'onEditRecipe');
-            const editRecipeButton = debugElement.query(By.css('#editRecipe'));
+        describe('buttons', () => {
+            let deleteRecipeBtn: DebugElement;
+            let toShoppingListBtn: DebugElement;
+            let editRecipeBtn: DebugElement;
```

Comment:
Für die Buttons wäre es glaube ich gut, diese in einzelne Describe-Blöcke zu schreiben, da sie nicht direkt zusammenhängen. Also für jeden Button einen eigenen Block -> `describe('deleteRecipeButton', ...)`